### PR TITLE
Support the alias of Git style sub-command

### PR DESCRIPTION
**Pull request by rike422**
this issues PR
#402 

## example

```js
var app = require('commander')
app
  .version('1.0.0')
  .command('foo', '...').alias('f')
  .command('bar', '...').alias('b')
  .command('baz', '...').alias('bz')
  .parse(process.argv)
```

### CODE REVIEW

**Comment by jamesr73** [comment]
I've noticed a related issue and I wonder if this PR (coincidentally) addresses it. I made the same change to the pm test fixture to give the install 'git-style command' an alias of 'i' and created the following simple test (logging help output because the test fails):

```javascript
var exec = require('child_process').exec
  , path = require('path')
  , should = require('should');

var bin = path.join(__dirname, './fixtures/pm')

exec(bin + ' help', function (error, stdout, stderr) {
  console.log('stdout:' + stdout)
  //stdout.should.containEql('install|i');
  //stdout.should.not.containEql('pm|i');
});
```

The help output has the alias against the main command not the subcommand:

```
  Usage: pm|i [options] [command]


  Commands:

    install [name]  install one or more packages
    search [query]  search with optional query
    cache           actions dealing with the cache
    list            list packages installed
    publish         publish or update package
    help [cmd]      display help for [cmd]

  Options:

    -h, --help     output usage information
    -V, --version  output the version number
```

When aliasing a command with associated action the help/usage output is as expected.

```javascript
var program = require('../')
  , should = require('should');

program
  .command('info [thing]')
  .alias('i')
  .action(function () {
  });

program
  .command('save [file]')
  .alias('s')
  .action(function() {
  });

program.parse(['node', 'test']);

program.help();
```

Outputs commands with aliases:

```
  Usage: test [options] [command]


  Commands:

    info|i [thing]
    save|s [file] 

  Options:

    -h, --help  output usage information
```

**Comment by jamesr73** [comment]
@rike422 I pulled from your fork and confirmed that it also addresses the issue with the help output. I'll send you a PR with the additional tests.

**Comment by zhiyelee** [comment]
Cool, thanks, can you fix the tests. @rike422

**Comment by rike422** [comment]
@jamesr73 glad to hear that it solves another issue,  I’ll merge it soon.

**Comment by rike422** [comment]
> Cool, thanks, can you fix the tests

@zhiyelee I think the reason it’s failing the tests is related to #550. If we solve that one then I think this PR will pass.

**Comment by zhiyelee** [comment]
yea, related to #550  
I like the feature and will review the code by this weekend. Thanks for the contribution.

**Comment by mikeerickson** [comment]
Yes please!!!! I was just wanting this very thing today!  Hope this gets accepted soon!

File: index.js [hunk 1]
```diff
@@ -850,8 +857,14 @@ Command.prototype.description = function(str) {
  */
 
 Command.prototype.alias = function(alias) {
-  if (0 == arguments.length) return this._alias;
-  this._alias = alias;
+  var command = undefined;
+  if(this.commands.length == 0) {
+    command = this;
+  } else {
+    command = this.commands[this.commands.length - 1]
```
Comment (zhiyelee) [inline index.js]: there may some flaw here.  Some of the commands may don't have an alias.
I have made some change as below, test failed.
image

Comment (rike422) [inline index.js]: @zhiyelee Thanks for the review, You should fix test in `test.command.executableSubcommandAlias.js`, I changed test as below and passed the test,
image

Comment (rike422) [inline index.js]: Should I add test case of that case?

Comment (zhiyelee) [inline index.js]: My bad, you are correct.
yes, please, thanks

Comment (zhiyelee) [inline index.js]: I think we can remove `bin + 'l'` test  from `test.command.executableSubcommandAlias.js`.

**Comment by zhiyelee** [comment]
Merged. Thanks @rike422  @jamesr73

**Pull request merged**